pkg/kubecost/asset.go

```diff
@@ -2838,6 +2838,31 @@ func (as *AssetSet) accumulate(that *AssetSet) (*AssetSet, error) {
 	return acc, nil
 }
 
+type Diff struct {
+	Asset Asset
+	ChangedType string
+}
+
+func DiffAsset(before, after *AssetSet) []Diff{
```

Comment:
Are you familiar with the concept of a "receiver" in Go? If not, read up about it and consider whether this should be a simple function like this, or something where an `AssetSet` is a receiver, with the function `Diff(that *AssetSet)`.

Reply:
@michaelmdresser and I thought it would make more sense for this function to be simple since we can ensure that the first argument is the earlier AssetSet (or AllocationSet when that is implemented) and the second is the later one.

Reply:
Sounds good!